index.js

```diff
@@ -114,7 +110,14 @@ function build (schema, options) {
         schemaKey = key + externalSchema.$id // relative URI
       }
 
-      if (ajvInstance.getSchema(schemaKey) === undefined) {
+      if (refResolver.getSchema(schemaKey) === undefined) {
+        refResolver.addSchema(externalSchema, key)
+      }
+
+      if (
+        ajvInstance.refs[schemaKey] === undefined &&
+        ajvInstance.schemas[schemaKey] === undefined
+      ) {
```

Comment:
I meant that this check should be enough to replace the new `RefResolver` holder as both (ajv and refresolver) are storing the same values

I will try it locally

Reply:
It won't work for nested schemas like 'subschema' and 'subschema#anchor'. I will work if we want ti know if Ajv has a schema, but we can't get it.

Reply:
Oh, I see. Ajv resolves references in that cases.

Another reason why I want to have a separate abstraction for this is because Ajv does two things: validates schemas at runtime and resolves references at compile time. It's not good. So in #504 I will have to create two Ajv instances (for schema validation and reference resolving), because there would be two different syntaxes.

And since we own this code it's easy to optimize it. For example in case if we want to reuse external schemas.

Reply:
@Eomm I rebased #504 to this branch. You can see there that RefResolver/Validator have different responsibilities and different reasons to be changed.